.gitignore

```diff
@@ -27,3 +27,10 @@ build/Release
 node_modules
 .gatsby-context.js
 .sass-cache/
+
+
+# Build
+public
```

Comment:
I think it'd be better to ignore `public/`.  
For dev, `gatsby develop` does everything you need.   
Whoever needs the production output should run a fresh `gatsby build`.